src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -219,20 +219,70 @@ internal static SyntaxToken ExtractAnonymousTypeMemberName(this ExpressionSyntax
             }
         }
 
-        internal static RefKind GetRefKind(this TypeSyntax syntax)
+        internal static RefKind GetRefKindInLocalOrReturn(this TypeSyntax syntax, BindingDiagnosticBag diagnostics)
         {
-            syntax.SkipRef(out var refKind);
+            syntax.SkipRefInLocalOrReturn(diagnostics, out var refKind);
             return refKind;
         }
 
-        internal static TypeSyntax SkipRef(this TypeSyntax syntax, out RefKind refKind)
+        /// <summary>
+        /// For callers that just want to unwrap a <see cref="RefTypeSyntax"/> and don't care if ref/readonly was there.
+        /// As these callers don't care about 'ref', they are in scenarios where 'ref' is not legal, and existing code
+        /// will error out for them.  Callers that do want to know what the ref-kind is should use <see
+        /// cref="SkipRefInLocalOrReturn(TypeSyntax, BindingDiagnosticBag, out RefKind)"/> so that an appropriate version check can be
+        /// performed to make sure using ref is legal in that language version.
+        /// </summary>
+        internal static TypeSyntax SkipRef(this TypeSyntax syntax)
+            => syntax.SkipRefInLocalOrReturn(diagnostics: null, out _);
```

Comment:
>SkipRefInLocalOrReturn [](http://example.com/codeflow?start=22&length=22)

It seems strange to be calling a method with a more specific name here. Also, ideally, we'd share code between `SkipRefInField()` and `SkipRefInLocalOrReturn()`.

Consider extracting a `private static TypeSyntax SkipRef(TypeSyntax syntax, BindingDiagnosticBag? diagnostics, out RefKind refKind)` that all the `SkipRef*` methods can call. (The helper could be `private` and not an extension method to prevent use outside of those methods.)

Reply:
wfm.

Reply:
done!